web/api/queries/trimmed_mean/trimmed_mean.c

```diff
@@ -0,0 +1,169 @@
+    struct grouping_trimmed_mean *g = (struct grouping_trimmed_mean *)r->internal.grouping_data;
+
+    NETDATA_DOUBLE value;
+    size_t available_slots = g->next_pos;
+
+    if(unlikely(!available_slots)) {
+        value = 0.0;
+        *rrdr_value_options_ptr |= RRDR_VALUE_EMPTY;
+    }
+    else if(available_slots == 1) {
+        value = g->series[0];
+    }
+    else {
+        sort_series(g->series, available_slots);
+
+        NETDATA_DOUBLE min = g->series[0];
+        NETDATA_DOUBLE max = g->series[available_slots - 1];
+
+        if (min != max) {
+            size_t slots_to_use = (size_t)((NETDATA_DOUBLE)available_slots * g->percent);
+            if(!slots_to_use) slots_to_use = 1;
+
+            NETDATA_DOUBLE percent_to_use = (NETDATA_DOUBLE)slots_to_use / (NETDATA_DOUBLE)available_slots;
+            NETDATA_DOUBLE percent_delta = g->percent - percent_to_use;
+
+            NETDATA_DOUBLE percent_interpolation_slot = 0.0;
+            NETDATA_DOUBLE percent_last_slot = 0.0;
+            if(percent_delta > 0.0) {
+                NETDATA_DOUBLE percent_to_use_plus_1_slot = (NETDATA_DOUBLE)(slots_to_use + 1) / (NETDATA_DOUBLE)available_slots;
+                NETDATA_DOUBLE percent_1slot = percent_to_use_plus_1_slot - percent_to_use;
+
+                percent_interpolation_slot = percent_delta / percent_1slot;
+                percent_last_slot = 1 - percent_interpolation_slot;
+            }
+
+            int start_slot, stop_slot, step, last_slot, interpolation_slot;
+            if(min >= 0.0 && max >= 0.0) {
+                start_slot = (int)((available_slots - slots_to_use) / 2);
+                stop_slot = start_slot + (int)slots_to_use;
+                last_slot = stop_slot - 1;
+                interpolation_slot = stop_slot;
+                step = 1;
+            }
+            else {
+                start_slot = (int)available_slots - 1 - (int)((available_slots - slots_to_use) / 2);
+                stop_slot = start_slot - (int)slots_to_use;
+                last_slot = stop_slot + 1;
+                interpolation_slot = stop_slot;
+                step = -1;
+            }
+
+            value = 0.0;
+            for(int slot = start_slot; slot != stop_slot ; slot += step)
+                value += g->series[slot];
+
+            size_t counted = slots_to_use;
+            if(percent_interpolation_slot > 0.0 && interpolation_slot >= 0 && interpolation_slot < (int)available_slots) {
+                value += g->series[interpolation_slot] * percent_interpolation_slot;
+                value += g->series[last_slot] * percent_last_slot;
+                counted++;
+            }
+
+            value = value / (NETDATA_DOUBLE)counted;
+
+            fprintf(stderr, "available_slot %zu, percent %f, slots_to_use %zu, start_slot %d, stop_slot %d, step %d, last_slot %d, interpolation_slot %d, percent_interpolation_slot %f, percent_last_slot %f, counted %zu\n",
```

Comment:
Is there a special reason to use `fprintf` instead `error` or `internal_error`?

Reply:
It was just left-over debug info. I removed it.